webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslationTextUnitFilterService.java

```diff
@@ -0,0 +1,100 @@
+package com.box.l10n.mojito.service.ai.translation;
+
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.entity.TMTextUnit;
+import jakarta.annotation.PostConstruct;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AITranslationTextUnitFilterService {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(AITranslationTextUnitFilterService.class);
+  private static final String HTML_TAG_REGEX = "<[^>]*>";
+  private static final Pattern HTML_TAG_PATTERN = Pattern.compile(HTML_TAG_REGEX);
+
+  protected Map<String, Pattern> excludePlaceholdersPatternMap;
+
+  @Autowired AITranslationFilterConfiguration aiTranslationFilterConfiguration;
+
+  public boolean isTranslatable(TMTextUnit tmTextUnit, Repository repository) {
+    boolean isTranslatable = true;
+
+    if (repository == null) {
+      logger.warn(
+          "Repository is null for text unit with id: {}, filtering will be skipped",
+          tmTextUnit.getId());
+      return isTranslatable;
+    }
+
+    if (aiTranslationFilterConfiguration.getRepositoryConfig() == null
+        || aiTranslationFilterConfiguration.getRepositoryConfig().get(repository.getName())
+            == null) {
+      logger.debug(
+          "No configuration found for repository: {}, filtering will be skipped",
+          repository.getName());
+      return isTranslatable;
+    }
+
+    AITranslationFilterConfiguration.RepositoryConfig repositoryConfig =
+        aiTranslationFilterConfiguration.getRepositoryConfig().get(repository.getName());
+
+    if (repositoryConfig.isExcludePlurals()) {
+      isTranslatable = !isPlural(tmTextUnit);
+    }
+
+    if (repositoryConfig.isExcludePlaceholders()) {
+      isTranslatable = isTranslatable && !containsPlaceholder(repository.getName(), tmTextUnit);
+    }
+
+    if (repositoryConfig.isExcludeHtmlTags()) {
+      isTranslatable = isTranslatable && !containsHtmlTag(tmTextUnit);
+    }
+
+    logger.debug(
+        "Text unit with name: {} should be translated: {}", tmTextUnit.getName(), isTranslatable);
```

Comment:
This log line might be hard to read since the text unit could be anything and there's no obvious delimiter here to know where the text unit name ends and the rest of the log begins
example

Text unit name is literrally "should be translated"
The log would be the following:
"Text unit with name: should be translated should be translated: true"
there's no comma or new line that helps the reader distinguish between what is the log message and what is the text unit itself.